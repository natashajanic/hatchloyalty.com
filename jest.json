{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
