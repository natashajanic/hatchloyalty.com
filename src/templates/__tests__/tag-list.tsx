import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import TagListTemplate, { ITagListTemplateProps } from '../tag-list'

describe('TagListTemplate', () => {
  let props: ITagListTemplateProps
  let mountedComponent: ShallowWrapper<any, any, TagListTemplate>

  beforeEach(() => {
    props = {
      pageContext: {
        basePath: '/blog/tag',
        tags: [
          {
            name: 'Test',
            postCount: 15,
          },
          {
            name: 'Press Release',
            postCount: 6,
          },
          {
            name: 'Enterprise Loyalty',
            postCount: 9,
          },
        ],
      },
    }
    mountedComponent = shallow(<TagListTemplate {...props} />)
  })

  describe('render', () => {
    it('renders a title for the page', async () => {
      expect(mountedComponent.find('h2').text()).toEqual('Tags')
    })

    it('renders each of the tags along with its post count', async () => {
      expect(mountedComponent.find(Link).length).toEqual(3)
      const testLink = mountedComponent.find(Link).at(0)
      expect(testLink.prop('to')).toEqual('/blog/tag/test/')
      expect(testLink.text()).toContain('Test')
      expect(testLink.text()).toContain('(15)')
      const pressReleaseLink = mountedComponent.find(Link).at(1)
      expect(pressReleaseLink.prop('to')).toEqual('/blog/tag/press-release/')
      expect(pressReleaseLink.text()).toContain('Press Release')
      expect(pressReleaseLink.text()).toContain('(6)')
      const enterpriseLoyaltyLink = mountedComponent.find(Link).at(2)
      expect(enterpriseLoyaltyLink.prop('to')).toEqual(
        '/blog/tag/enterprise-loyalty/'
      )
      expect(enterpriseLoyaltyLink.text()).toContain('Enterprise Loyalty')
      expect(enterpriseLoyaltyLink.text()).toContain('(9)')
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
