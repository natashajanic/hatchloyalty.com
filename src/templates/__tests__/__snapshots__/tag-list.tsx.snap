// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagListTemplate render matches the expected snapshot 1`] = `
<Layout
  pageStyle="offWhite"
>
  <SEO
    title="All Tags"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="main"
    m="0 auto"
    maxWidth="1024px"
    px={
      Array [
        3,
        4,
      ]
    }
  >
    <Styled(Component)
      blacklist={
        Array [
          "css",
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "cursor",
        ]
      }
      border="1px solid"
      borderColor="grayLight"
      borderRadius={4}
      is={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "Body": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-iwsKbI",
              "isStatic": false,
              "rules": Array [
                [Function],
                "max-height: 415px; overflow: scroll;",
              ],
            },
            "defaultProps": Object {
              "blacklist": Array [
                "css",
              ],
              "is": "div",
            },
            "displayName": "Styled(Component)",
            "foldedComponentIds": Array [],
            "propTypes": Object {},
            "render": [Function],
            "styledComponentId": "sc-iwsKbI",
            "systemComponent": true,
            "target": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
          "Footer": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-gZMcBi",
              "isStatic": false,
              "rules": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
            "defaultProps": Object {
              "alignItems": "center",
              "blacklist": Array [
                "css",
                "alignItems",
                "border",
                "borderTop",
                "borderRight",
                "borderBottom",
                "borderLeft",
                "borderColor",
                "display",
                "justifyContent",
                "m",
                "mt",
                "mr",
                "mb",
                "ml",
                "mx",
                "my",
                "p",
                "pt",
                "pr",
                "pb",
                "pl",
                "px",
                "py",
              ],
              "borderColor": "grayLight",
              "borderTop": 1,
              "display": "flex",
              "is": "footer",
              "justifyContent": "space-between",
              "mt": 4,
              "mx": -3,
              "pt": 3,
              "px": 3,
            },
            "displayName": "Panel.Footer",
            "foldedComponentIds": Array [],
            "propTypes": Object {
              "alignItems": [Function],
              "border": [Function],
              "borderBottom": [Function],
              "borderColor": [Function],
              "borderLeft": [Function],
              "borderRight": [Function],
              "borderTop": [Function],
              "display": [Function],
              "justifyContent": [Function],
              "m": [Function],
              "mb": [Function],
              "ml": [Function],
              "mr": [Function],
              "mt": [Function],
              "mx": [Function],
              "my": [Function],
              "p": [Function],
              "pb": [Function],
              "pl": [Function],
              "pr": [Function],
              "pt": [Function],
              "px": [Function],
              "py": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-gZMcBi",
            "systemComponent": true,
            "target": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
          "Header": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": false,
              "rules": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
            "defaultProps": Object {
              "alignItems": "center",
              "blacklist": Array [
                "css",
                "alignItems",
                "border",
                "borderTop",
                "borderRight",
                "borderBottom",
                "borderLeft",
                "borderColor",
                "display",
                "justifyContent",
                "m",
                "mt",
                "mr",
                "mb",
                "ml",
                "mx",
                "my",
                "p",
                "pt",
                "pr",
                "pb",
                "pl",
                "px",
                "py",
                "flex",
                "flexDirection",
              ],
              "borderBottom": 1,
              "borderColor": "grayLight",
              "display": "flex",
              "is": "header",
              "justifyContent": "space-between",
              "mb": 4,
              "mx": -3,
              "pb": 3,
              "px": 3,
            },
            "displayName": "Panel.Header",
            "foldedComponentIds": Array [],
            "propTypes": Object {
              "alignItems": [Function],
              "border": [Function],
              "borderBottom": [Function],
              "borderColor": [Function],
              "borderLeft": [Function],
              "borderRight": [Function],
              "borderTop": [Function],
              "display": [Function],
              "flex": [Function],
              "flexDirection": [Function],
              "justifyContent": [Function],
              "m": [Function],
              "mb": [Function],
              "ml": [Function],
              "mr": [Function],
              "mt": [Function],
              "mx": [Function],
              "my": [Function],
              "p": [Function],
              "pb": [Function],
              "pl": [Function],
              "pr": [Function],
              "pt": [Function],
              "px": [Function],
              "py": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "systemComponent": true,
            "target": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
          "Title": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-dnqmqq",
              "isStatic": false,
              "rules": Array [
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
                [Function],
              ],
            },
            "defaultProps": Object {
              "blacklist": Array [
                "css",
                "color",
                "bg",
                "fontSize",
                "m",
                "mt",
                "mr",
                "mb",
                "ml",
                "mx",
                "my",
                "p",
                "pt",
                "pr",
                "pb",
                "pl",
                "px",
                "py",
                "alignItems",
                "display",
              ],
              "color": "offBlack",
              "fontSize": 4,
              "is": "h4",
              "m": 0,
            },
            "displayName": "Styled(Component)",
            "foldedComponentIds": Array [],
            "propTypes": Object {
              "alignItems": [Function],
              "bg": [Function],
              "color": [Function],
              "display": [Function],
              "fontSize": [Function],
              "m": [Function],
              "mb": [Function],
              "ml": [Function],
              "mr": [Function],
              "mt": [Function],
              "mx": [Function],
              "my": [Function],
              "p": [Function],
              "pb": [Function],
              "pl": [Function],
              "pr": [Function],
              "pt": [Function],
              "px": [Function],
              "py": [Function],
            },
            "render": [Function],
            "styledComponentId": "sc-dnqmqq",
            "systemComponent": true,
            "target": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-gzVnrw",
            "isStatic": false,
            "rules": Array [
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              "box-shadow: #dfdfdf 0 3px 10px;",
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
              [Function],
            ],
          },
          "defaultProps": Object {
            "blacklist": Array [
              "cursor",
            ],
            "border": "1px solid",
            "borderColor": "grayLight",
            "borderRadius": 4,
            "m": 0,
            "p": 3,
            "width": "100%",
          },
          "displayName": "Panel",
          "foldedComponentIds": Array [],
          "propTypes": Object {
            "alignItems": [Function],
            "border": [Function],
            "borderBottom": [Function],
            "borderColor": [Function],
            "borderLeft": [Function],
            "borderRadius": [Function],
            "borderRight": [Function],
            "borderTop": [Function],
            "display": [Function],
            "flex": [Function],
            "flexDirection": [Function],
            "height": [Function],
            "justifyContent": [Function],
            "m": [Function],
            "mb": [Function],
            "ml": [Function],
            "mr": [Function],
            "mt": [Function],
            "mx": [Function],
            "my": [Function],
            "p": [Function],
            "pb": [Function],
            "pl": [Function],
            "position": [Function],
            "pr": [Function],
            "pt": [Function],
            "px": [Function],
            "py": [Function],
            "width": [Function],
          },
          "render": [Function],
          "styledComponentId": "sc-gzVnrw",
          "systemComponent": true,
          "target": Object {
            "$$typeof": Symbol(react.forward_ref),
            "render": [Function],
          },
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      m={0}
      my={4}
      p={3}
      px={4}
      py={4}
      width="100%"
    >
      <h2>
        Tags
      </h2>
      <ul>
        <li
          key="Test"
        >
          <ForwardRef
            to="/blog/tag/test/"
          >
            Test
             (
            15
            )
          </ForwardRef>
        </li>
        <li
          key="Press Release"
        >
          <ForwardRef
            to="/blog/tag/press-release/"
          >
            Press Release
             (
            6
            )
          </ForwardRef>
        </li>
        <li
          key="Enterprise Loyalty"
        >
          <ForwardRef
            to="/blog/tag/enterprise-loyalty/"
          >
            Enterprise Loyalty
             (
            9
            )
          </ForwardRef>
        </li>
      </ul>
    </Styled(Component)>
  </Styled(Component)>
</Layout>
`;
